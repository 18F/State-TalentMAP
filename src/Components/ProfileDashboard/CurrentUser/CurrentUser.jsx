import React from 'react';
import { USER_PROFILE } from '../../../Constants/PropTypes';
import { NO_EMAIL, NO_USER_SKILL_CODE } from '../../../Constants/SystemMessages';
import SectionTitle from '../SectionTitle';
import InformationDataPoint from '../InformationDataPoint';
import Status from './Status';
import EditProfile from './EditProfile';
import ProfilePicture from '../../ProfilePicture';
import StaticDevContent from '../../StaticDevContent/StaticDevContent';

const CurrentUser = ({ userProfile }) => (
  <div className="usa-grid-full current-user">
    <div className="current-user-top">
      <div className="section-padded-inner-container">
        <StaticDevContent>
          <Status />
        </StaticDevContent>
        <ProfilePicture />
        <StaticDevContent>
          <EditProfile />
        </StaticDevContent>
        <SectionTitle small title={`${userProfile.user.first_name} ${userProfile.user.last_name}`} className="current-user-name" />
        <InformationDataPoint
          content={userProfile.skill_code || NO_USER_SKILL_CODE}
        />
      </div>
    </div>
    <div className="current-user-bottom">
      <div className="section-padded-inner-container">
        <SectionTitle small title="Contact Information" />
        <InformationDataPoint title="Email Address" content={userProfile.user.email || NO_EMAIL} />
        <StaticDevContent>
          <InformationDataPoint title="Office Number" content="+301-779-0379 ext. 3" />
        </StaticDevContent>
        <StaticDevContent>
          <InformationDataPoint title="Personal Contact Number" content="+240-331-7189" />
        </StaticDevContent>
        <StaticDevContent>
          <InformationDataPoint
            title="Post/Office Address"
            content="1234 Washington St. NW, Washington, DC 20009"
          />
        </StaticDevContent>
      </div>
    </div>
  </div>
);

CurrentUser.propTypes = {
  userProfile: USER_PROFILE.isRequired,
};

export default CurrentUser;
