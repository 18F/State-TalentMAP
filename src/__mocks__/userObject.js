const userObject = {
  skill_code: 'INFORMATION MANAGEMENT (2880)',
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
  },
  is_cdo: true,
};

export default userObject;
