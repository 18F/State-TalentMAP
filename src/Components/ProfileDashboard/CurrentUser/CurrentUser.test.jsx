import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import userObject from '../../../__mocks__/userObject';
import CurrentUser from './CurrentUser';

describe('CurrentUserComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<CurrentUser userProfile={userObject} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot when editor is shown', () => {
    const wrapper = shallow(<CurrentUser userProfile={userObject} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
