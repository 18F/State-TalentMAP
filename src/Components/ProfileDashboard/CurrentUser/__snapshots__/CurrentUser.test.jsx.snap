// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentUserComponent matches snapshot 1`] = `
<div
  className="usa-grid-full current-user"
>
  <CurrentUserGeneralInformation
    showEditLink={true}
    useGroup={false}
    userProfile={
      Object {
        "is_cdo": true,
        "skill_code": "INFORMATION MANAGEMENT (2880)",
        "user": Object {
          "email": "doej@state.gov",
          "first_name": "John",
          "last_name": "Doe",
        },
      }
    }
  />
  <CurrentUserContactInformation
    showEditLink={true}
    userProfile={
      Object {
        "is_cdo": true,
        "skill_code": "INFORMATION MANAGEMENT (2880)",
        "user": Object {
          "email": "doej@state.gov",
          "first_name": "John",
          "last_name": "Doe",
        },
      }
    }
  />
</div>
`;
